aws/rust-runtime/aws-config/src/default_provider/credentials.rs

```diff
@@ -161,9 +161,9 @@ impl Builder {
     }
 
     /// Override the configuration used for this provider
-    pub fn configure(mut self, config: ProviderConfig) -> Self {
-        self.region_chain = self.region_chain.configure(&config);
-        self.conf = Some(config);
+    pub fn configure(mut self, config: &ProviderConfig) -> Self {
+        self.region_chain = self.region_chain.configure(config);
+        self.conf = Some(config.clone());
```

Comment:
I know _in general_, we have [this API guideline](https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control) where if the implementation ends up using an owned value, the function/method should take as input an owned value from the beginning, rather than taking it by reference and internally clone it. What do we think?